src/functions-and-arrays.js

```diff
@@ -1,42 +1,99 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+
+function maxOfTwoNumbers( n1, n2) {
+  if (n1>n2) {
+    return n1;
+  }  else if (n2>n1) {
+     return n2;
+  } else {
+    return n1,n2;
+  }
+}
+
+let n1 = 9;
+let n2 = 10;
+console.log(maxOfTwoNumbers(1,2));
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
-
+function findLongestWord(longWord) {
+  let emptyStr = "";
+  if(!longWord.length) {
+    return null;
+  } else if (longWord.length === 1) {
+    return longWord[0];
+  } else {
+    for ( let i = 0;  i < longWord.length ; i++) {
+      if ( longWord[i].length > emptyStr.length) {
+        emptyStr = longWord[i];
+      }
+    } return emptyStr;
+  }
+}
 
 
 // Iteration #3: Calculate the sum
-const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+//3.1
 
-function sumNumbers() {}
+const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(arr) {
+  let total = 0;
+  if (arr.length === 0) {
+     total = 0;
+  } else { 
+    for (let i = 0; i < arr.length; i++) {
+      total += arr[i];
+    }
+  } return total;
+}
 
+ 
+// Iteration #3.2 Bonus:
 
-// Iteration #3.1 Bonus:
-function sum() {}
 
+function sum(sumNumbers) {
+  
+}
 
 
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
-
-
+function averageNumbers(arr) {
+  let total = 0; 
+  for (let i = 0; i < arr.length; i++) {
+    total += arr[i];
+  } if (arr === []) {
```

Comment:
Here, you can't compare the argument (arr) to that empty array '[]' as it will not work. To check if an empty array is called, you can use: `if (arr.length === 0) return null;` or `if (!arr.length) return null`